src/System.Windows.Forms/src/System/Windows/Forms/ToolStripSplitStackLayout.cs

```diff
@@ -89,12 +86,7 @@ private void CalculatePlacementsHorizontal()
                     // if we have something set to overflow always we need to show an overflow button
                     if (item.Overflow == ToolStripItemOverflow.Always)
                     {
-#if DEBUG
-                        if (DebugLayoutTraceSwitch.TraceVerbose)
-                        {
-                            Debug.WriteLine(string.Format(CultureInfo.CurrentCulture, "OverflowRequired - item set to always overflow: {0} ", item));
-                        }
-#endif
+                        DebugLayoutTraceSwitch.TraceVerbose($"OverflowRequired - item set to always overflow: {item} ");
```

Comment:
This is actually a deoptimisation. I made this change to make the code more consistent, and to put this up for debate because this pattern is used across the whole codebase.
Previously, the message string was created only when tracing was set to verbose. After this change it's going to be created whenever we're in debug, as [TraceVerbose](https://github.com/dotnet/winforms/blob/a8e341cb99955ec3cbdea9815dfa71e487661633/src/Common/src/TraceSwitchExtensions.cs#L12) doesn't handle interpolated strings.
I see 2 potential fixes to this pattern:

1. Introduce a custom interpolated string handler for the `TraceVerbose` extension method
2. inline uses of the extension method as such:
```suggestion
                        Debug.WriteLineIf(DebugLayoutTraceSwitch!.TraceVerbose, $"SendNextItemToOverflow to free space for {item}");
```

Reply:
The reason we had introduced the extension method was to avoid using the null-forgiving operator. You can check https://github.com/dotnet/winforms/pull/8072#discussion_r1009826425 for more details.

Reply:
#8719

Reply:
Your proposal, @halgab, is currently under discussion and received positive feedback. Would you like to open the draft for review and potentially utilize it in this PR?

Reply:
Sure!